### PR TITLE
Rewrite DoNotUseInsecureXSLTScriptExecutionAnalyzer

### PR DESCRIPTION
**Pull request by Evangelink**
Contributes to #2103

### CODE REVIEW

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureXSLTScriptExecution.cs [hunk 1]
```diff
@@ -1,294 +1,319 @@
...
+                isDataflowRule: false);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleDoNotUseInsecureXSLTScriptExecution);
 
 #pragma warning disable RS1026 // Enable concurrent execution
         public override void Initialize(AnalysisContext analysisContext)
 #pragma warning restore RS1026 // Enable concurrent execution
         {
-            // TODO: Make analyzer thread-safe.
             //analysisContext.EnableConcurrentExecution();
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/DoNotUseInsecureXSLTScriptExecution.cs:37]: I could not find how to rewrite the code to allow concurrent execution. We currently rely on the analysis of variable declarator and assignments before analyzing invocations or object creations.

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SecurityDiagnosticHelpers.cs [hunk 2]
```diff
@@ -12,14 +12,12 @@ namespace Microsoft.NetFramework.Analyzers.Helpers
 {
     public static class SecurityDiagnosticHelpers
     {
-        public static bool IsXslCompiledTransformLoad([NotNullWhen(returnValue: true)] IMethodSymbol? method, CompilationSecurityTypes xmlTypes)
-        {
-            return method != null
+        public static bool IsXslCompiledTransformLoad([NotNullWhen(returnValue: true)] this IMethodSymbol? method, CompilationSecurityTypes xmlTypes)
```
Comment (Evangelink) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SecurityDiagnosticHelpers.cs:15]: Refactoring to use as extension method + use of expression body is done in a second commit.

**Comment by Evangelink** [comment]
@dotpaul for review

File: src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SecurityDiagnosticHelpers.cs [hunk 3]
```diff
@@ -263,6 +245,30 @@ public static string GetNonEmptyParentName(SyntaxNode current, SemanticModel mod
             return string.Empty;
         }
 
+        /// <summary>
+        /// Get non-empty class or method name which encloses the current syntax node
```
Comment (dotpaul) [inline src/NetAnalyzers/Core/Microsoft.NetFramework.Analyzers/Helpers/SecurityDiagnosticHelpers.cs:249]:
```suggestion
        /// Get class or method name which encloses the current symbol node
```

**Comment by Evangelink** [comment]
@dotpaul @mavasani could you investigate or ask someone to investigate these build time failures? They seem to happen more and more.

**Comment by mavasani** [comment]
Seems like the tests are timing out intermittently, so wondering if this is a product or test race condition. Unfortunately, we are not collecting any dumps on the published build artifacts, so it is not easy to determine the failure cause. I'll try to get a repro locally.

**Review by dotpaul — APPROVED** [review]
:ship: Thanks @Evangelink !

**Pull request merged**